Flank/src/main/java/com/walmart/otto/filter/TestFilters.java

```diff
@@ -0,0 +1,121 @@
+package com.walmart.otto.filter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class TestFilters {
+
+  private static final Pattern FILTER_ARGUMENT =
+      Pattern.compile(
+          "(not)?(annotation|class|package|testfile)\\s+(.+)", Pattern.CASE_INSENSITIVE);
+
+  public static TestFilter fromCommandLineArguments(String arguments) {
+    String[] args = arguments.split(";");
+
+    List<TestFilter> filters =
+        Arrays.stream(args)
+            .map(String::trim) //splitting by semicolon will leave leading or trailing spaces
+            .map(TestFilters::parseSingleFilter)
+            .collect(Collectors.toList());
+
+    Collections.addAll(filters, notIgnored());
+
+    return allOf(filters);
+  }
+
+  public static TestFilter createDefault() {
+    return notIgnored();
+  }
+
+  private static TestFilter parseSingleFilter(String argument) {
+    Matcher matcher = FILTER_ARGUMENT.matcher(argument);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException("Invalid argument: " + argument);
+    }
+
+    boolean isNegate = matcher.group(1) != null;
+
+    Collection<String> args =
```

Comment:
To make the api consistent, I would force no space in group(3).
Basically:
if group(3) contains \s, `throw IllegalArgumentException("space in " + group(3))`

Reply:
To be fair, it still wouldn't be super-consistent. In AndroidJUnitRunner you have in order to specify multiple options you have to declare multiple `-e` whereas in Flank you pass you just one -f and you wrap everything with quotes. As I said before, this is just inspired by AndroidJUnitRunner, it doesn't aim to be 100% compatible :)

Reply:
I mean consistent against Flank itself. 
I don't like the fact that `package x, y` might have issues and `annotation x, y` will not.

I would remove the possibility of spaces and be done with it.

Reply:
>I mean consistent against Flank itself.
I don't like the fact that package x, y might have issues and annotation x, y will not.

At the moment you are free to add spaces regardless the operator. The test covers just the annotation case but it works at the same way with every operator.